docs/DeveloperGuide.md

```diff
@@ -322,7 +327,74 @@ The implementation of `Model#unlinkFriend()` is as follows:
+* available during the specified time (hour and day) based on schedule
+* are linked to the specified game 
+* and are ordered by the highest skill value
+
+This command allows the user to easily find the highest-skilled available friends who play the specified game to coordinate 
+gaming sessions with from the recommendations friends list displayed after the command is executed. 
+
+#### Implementation: 
+
+To help you understand how the recommendation functionality is implemented, the activity diagram below summarises the actions 
+gitGud undergoes to display the recommendations friends list to the user: 
+
+<img src="images/RecommendCommandActivityDiagram.png" width="1000" />
+
+<ins>Step 1: Parsing and triggering recommend command execution</ins><br>
+
+Similar to [delete](#delete-feature) and [link](#link-feature) features above, the parse and execute actions shown in the activity 
+diagram are implemented via invoking`RecommendCommandParser#parse(String)`, which extracts a specified game and timing from the 
+user input and constructs a `RecommendCommand` instance with the extracted data. 
+
+Then, the `RecommendCommand#execute(model)` method is invoked by an instance of the `LogicManager` class which uses the extracted
+data to produce a list of friend recommendations to the user.
+
+<ins>Step 2: Filtering and sorting the friends list to get recommendations friends list</ins><br> 
+
+The sequence diagram below illustrates what happens during the `RecommendCommand#execute(model)` method to produce the desired list of friend recommendations:
+
+<img src="images/RecommendCommandSequenceDiagram1.png" width="1000" /><br> 
+
+As shown, the recommend feature constructs an instance of `FriendRecommendFilterPredicate` which returns true
+if the friend plays the specified game and whose schedule is available during the specified timing based on the extracted data. 
+It also constructs an instance of `Comparator` which is used to sort friends in order of the highest skill value for the specified game.
+
+Next, the continued sequence diagram below shows the `ModelManager#updateFilteredAndSortedFriendsList(Predicate, Comparator)`
+setting the predicate of `FilteredList` and comparator of `SortedList` instances managed by the `ModelManager` with 
+the new `FriendRecommendFilterPredicate` and `Comparator` instances, which causes filtering and sorting of the list and 
+produces the list of friend recommendations. 
+
+<img src="images/RecommendCommandSequenceDiagram2.png" width="1000" />
+
+<ins>Step 3: Displaying the recommended friends</ins> 
+
+Due to the use of JavaFX's `FilteredList` and `SortedList`, which listens for and tracks for changes whenever the
+`FilteredList#setPredicate(Predicate)` or `SortedList#setComparator(Comparator)` methods are invoked, the user interface is
+notified and updated to display the filtered and sorted friends list based on the produced list of friend recommendations in step 2. 
```

Comment:
A bit confusing as to which list the JavaFX-based `UI` is observing. There has been no mention of the term `ObservableList`, which is the main reason the `UI` is being updated whenever there's a change to the underlying `SortedList<Friend>`, which is built on top of the `FilteredList<Friend>`, which in turn is built on top of a `FriendsList`.